base/inference.jl

```diff
@@ -734,6 +747,172 @@ function limit_type_depth(t::ANY, d::Int, cov::Bool, vars::Vector{TypeVar}=TypeV
     return R
 end
 
+function limit_type_size(t::ANY, compare::ANY, source::ANY)
+    source = svec(unwrap_unionall(compare), unwrap_unionall(source))
+    source[1] === source[2] && (source = svec(source[1]))
+    type_more_complex(t, compare, source) || return t
+    r = _limit_type_size(t, compare, source)
+    #@assert !isa(t, Type) || t <: r
```

Comment:
perhaps. currently it would just trigger the same false-bug detection as https://github.com/JuliaLang/julia/pull/21907